### PR TITLE
ORC : Avoid modifying existing conf of HadoopOutputFile rather create new one

### PR DESCRIPTION
**Pull request by singhpk234**
A probable fix to 

Close https://github.com/apache/iceberg/issues/4383

### CODE REVIEW

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java [hunk 1]
```diff
@@ -162,11 +162,9 @@ private WriteBuilder createContextFunc(Function<Map<String, String>, Context> ne
     public <D> FileAppender<D> build() {
       Preconditions.checkNotNull(schema, "Schema is required");
 
-      Configuration conf;
+      Configuration conf = new Configuration();
       if (file instanceof HadoopOutputFile) {
-        conf = ((HadoopOutputFile) file).getConf();
-      } else {
-        conf = new Configuration();
+        ((HadoopOutputFile) file).getConf().forEach(entry -> conf.set(entry.getKey(), entry.getValue()));
```
Comment (openinx) [inline orc/src/main/java/org/apache/iceberg/orc/ORC.java:167]: Seems we will merge the hadoop file's configuration with the default environment configuration by default if it's a `HadoopOutputFile`, right ? 

Should we just use the original code piece ( according to [here](https://github.com/apache/iceberg/pull/4291/files#diff-34d7fce4c1d9417fa9342247cf5ace0636cd86591efbed0555ae80197f79303dL90-L94)):

```java
      if (file instanceof HadoopOutputFile) {
        this.conf = new Configuration(((HadoopOutputFile) file).getConf());
      } else {
        this.conf = new Configuration();
      }
```

Comment (singhpk234) [reply orc/src/main/java/org/apache/iceberg/orc/ORC.java]: makes sense, made the changes

**Review by openinx — APPROVED** [review]
Looks good to me, thanks @singhpk234 for the contribution !

**Comment by singhpk234** [comment]
Thanks @openinx , for prompt review :) !!!!

**Pull request merged**